src/runner_metrics.py

```diff
@@ -225,7 +317,52 @@ def _clean_up_shared_fs(fs: shared_fs.SharedFilesystem) -> None:
         logger.exception("Could not delete shared filesystem for runner %s.", fs.runner_name)
 
 
-def extract(flavor: str, ignore_runners: set[str]) -> IssuedMetricEventsStats:
+def _extract_fs(
+    runner_fs: shared_fs.SharedFilesystem, flavor: str, gh_api: GhApi
+) -> IssuedMetricEventsStats:
+    """Extract and issue metrics from a shared filesystem.
+
+    Args:
+        runner_fs: The shared filesystem for a specific runner.
+        flavor: The flavor of the runner.
+        gh_api: The GitHub API client.
```

Comment:
The name of this function is related to the shared filesystem, is it the right function to put the call to GitHub for additional data?

Reply:
Updated the code. The duration is now calculated in a separate module.